### ISSUE TITLE
Hot Exit via Unity Launcher

### ISSUE THREAD

**Issue by bzier** [Microsoft/vscode#46438]
Issue Type: <b>Bug</b>

When multiple editors are open, each with unsaved files, choosing File->Exit (Ctrl-Q) will do a 'hot exit'. Launching VS Code again will restore all editors along with all the unsaved files. However, when right-clicking the Unity launcher icon for VS Code and choosing Quit, each of the editors prompts to save its unsaved files.

**Expected:**
An overall exit/quit, whether via the File menu or the Unity launcher, should behave consistently and perform the 'hot exit'.

**Actual:**
The behavior of the exit varies depending on how it was invoked.

VS Code version: Code 1.21.1 (79b44aa704ce542d8ca4a3cc44cfca566e7720f1, 2018-03-14T14:46:30.761Z)
OS version: Linux x64 4.4.0-116-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Athlon(tm) 64 X2 Dual Core Processor 5000+ (2 x 2600)|
|Load (avg)|2, 2, 1|
|Memory (System)|5.83GB (0.17GB free)|
|Process Argv|/usr/share/code/code --unity-launch|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none
Reproduces without extensions

**Comment by Tyriar** [comment]
I'm not sure we can do anything about this, I think Unity is probably telling each window to close as opposed to telling all windows to close (which I guess it can't communicate). I assume this could also be reproduced in Windows but not macOS as quitting the application (all windows) is an OS-level concept.

**Comment by Tyriar** [comment]
@bpasero I say we should close this off as not fixed unless you think it's a good idea to put some delay before we send the save files dialog to check how many windows are being closed.

**Comment by bzier** [comment]
Fair enough, thanks for the quick response and insight into the problem. TBH, I didn't even realize there was a behavioral difference between `right-click->Quit` and `File->Exit`. I did a search to see if the issue had been reported for closing multiple editors as opposed to just one and I found #101, which indicated that hot-exit worked for multiple editors too. Now that I know `File->Exit` works, I'll probably just use that instead.
